atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/TimestampRangeDelete.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.api;
+
+import org.immutables.value.Value.Immutable;
+
+@Immutable
+public interface TimestampRangeDelete {
+    long timestamp();
+    boolean endInclusive();
+    boolean deleteSentinels();
+
+    default long maxTimestampToDelete() {
+        return timestamp() + (endInclusive() ? 0 : -1);
+    }
+
+    default long minTimestampToDelete() {
+        return deleteSentinels() ? Value.INVALID_VALUE_TIMESTAMP : Value.INVALID_VALUE_TIMESTAMP + 1;
+    }
```

Comment:
nit: it looks a bit unexpected to not have `@Value.Lazy` annotations on these methods; as written I think we end up re-doing the calculations repeatedly (they're cheap, but still no reason to redo them).